drivers/backup/portworx/portworx.go

```diff
@@ -422,28 +679,145 @@ func (p *portworx) WaitForBackupLocationDeletion(
 	return nil
 }
 
+func (p *portworx) GetBackupLocationUID(ctx context.Context, orgID string, backupLocationName string) (string, error) {
+	backupLocationEnumerateReq := &api.BackupLocationEnumerateRequest{
+		OrgId: orgID,
+	}
+	enumerateRsp, err := p.EnumerateBackupLocation(ctx, backupLocationEnumerateReq)
```

Comment:
As you noted, using `EnumerateBackupLocation` instead of `EnumerateBackupLocationByUser` can cause issues when multiple backup locations have the same name. This is also true for other enumeration methods. Consider a case in which the user is deleted and an admin wants to remove that backup location created by the user, using `EnumerateBackupLocationByUser` would not work because the admin did not create it, making it impossible to get the UID and delete the backup location. I am still unsure how to handle duplicate names unless we pass the owner ID or the actual owner context.

Reply:
@mkoppal-px 
Can you please suggest?
It is a corner case and we do not have any test cases with same name backup location for two users.
We can add a note so that we do not run into any issue in future?